lib/system.nim

```diff
@@ -2965,14 +2965,12 @@ when notJSnotNims and not defined(nimSeqsV2):
       assert y == "abcgh"
     discard
 
-proc arrayWith*[T](y: T, size: static int): array[size, T] {.raises: [].} =
+proc arrayWith*[T](y: T, size: static int): array[size, T] {.noinit, raises: [].} =
```

Comment:
this smells unsafe when T is `ref` or similar

Reply:
Good point actually, I might need to disable destructors as well.